src/coreclr/gc/gc.cpp

```diff
@@ -25170,6 +25230,14 @@ void gc_heap::check_heap_count ()
         dynamic_heap_count_data.space_cost_increase_per_step_up   = space_cost_increase_per_step_up;
         dynamic_heap_count_data.space_cost_decrease_per_step_down = space_cost_decrease_per_step_down;
 
+        GCEventFireDynamicHeapCountData(
```

Comment:
it would be helpful to add the GC index (for easy correlation), smoothed value and the new # of heaps.